### ISSUE TITLE
Average processing time not correct in v0.107.29 since update from v0.107.28

### ISSUE THREAD

**Issue by ev4x** [AdguardTeam/AdGuardHome#5756]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

Linux, Other (please mention the version in the description)

### CPU architecture

x86

### Installation

GitHub releases or script from README

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

v0.107.29

### Description

I use Optimitic caching my average processing time was about 2ms. Then i update from v0.107.28 to v0.107.29 now my average processing time is between 20 and 30ms.

There must be something wrong in calculation. I look at query log and optimistic caching work like before the update. But calculation of v0.107.29 must be wrong

**Comment by EugeneOne1** [comment]
@ev4x, hello and thanks for the report.  Could you please refine the exact upstream protocols used in your setup?  And also, if the issue persists with optimistic caching disabled?

**Comment by ev4x** [comment]
> @ev4x, hello and thanks for the report. Could you please refine the exact upstream protocols used in your setup? And also, if the issue persists with optimistic caching disabled?

You mean my upstream configuration? I see most is cache under 1ms normal answers are 20 to 35ms. I changed nothing average processing time was 2ms than update and now 22ms which is not possible because most is cache

**Comment by EugeneOne1** [comment]
@ev4x, yeah, I'm asking for upstream configuration, the URL schemas in particular (the part before `://`), or if those are plain IP addresses?  The increasing of processing time could actually be caused by any change in the DNS handling pipeline, but currently we primarily suspect our recent refactoring of upstream bootstrap logic.

**Comment by ev4x** [comment]
> @ev4x, yeah, I'm asking for upstream configuration, the URL schemas in particular (the part before `://`), or if those are plain IP addresses? The increasing of processing time could actually be caused by any change in the DNS handling pipeline, but currently we primarily suspect our recent refactoring of upstream bootstrap logic.

Its cloudflare i changed nothing 
tls://1.1.1.1
tls://[2606:4700:4700::1111]
tls://1.0.0.1
tls://[2606:4700:4700::1001]
https://1.1.1.1/dns-query
https://[2606:4700:4700::1111]/dns-query
https://1.0.0.1/dns-query
https://[2606:4700:4700::1001]/dns-query

**Comment by ev4x** [comment]
@EugeneOne1 i think i have the fault it was some issue of a blocklist. I resetet the stats. Then done some adblock test with many blocked domains so my average processing time was over 3000ms so i know its related to blocking. Then I deleted all blocklist and added them new and now it works again average processing time is back to 3ms

**Comment by ev4x** [comment]
I close it now average time is 1ms so it was related too some blocklist maybe damaged idk after deleting and readding all lists everything is normal like in last version

**Comment by ev4x** [comment]
I need to reopen it was 2 days good at 1ms now its back to 39ms so something is wrong didn't want to delete and readd all lists again i think its related to blocking or blocklist but the times in querry log all fine for blocking under 0.1 ms or just calculations

**Comment by ev4x** [comment]
@EugeneOne1 i dont get it fixed still seconds of average processing time today i found a strange thing a response time with minus how is this possible could this be a reason?
![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/129671265/cb89c874-9856-44be-83ac-7ce3f4432fb8)

**Comment by ev4x** [comment]
@EugeneOne1 i dont get it fixed i am now on v0.107.33 have seconds of average processing time after some time >2000ms today i found a strange thing a response time with minus how is this possible?

![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/129671265/cb89c874-9856-44be-83ac-7ce3f4432fb8)

And what i also one time found is this

![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/129671265/2eeb645e-f101-4d62-9151-198ab5c377e9)

**Comment by ev4x** [comment]
@EugeneOne1 i dont get it fixed i am now on v0.107.33 and have seconds of average processing time after some time >2000ms today i found a strange thing a response time with minus how is this possible?

![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/129671265/cb89c874-9856-44be-83ac-7ce3f4432fb8)

And what i also one time found is this

![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/129671265/2eeb645e-f101-4d62-9151-198ab5c377e9)

**Comment by ev4x** [comment]
@ainar-g @EugeneOne1 

I think i have the fault for this i cleared stats and had about 156 queries and my response time was  
          
**21694** **ms** 

Than i checked every query including the blocked ones and every query was complety normal mostly cache only one query i found with -0,02ms was not normal. I think this is the issue for this high response time and i mean a negative query is not possible and may result in a wrong calculation.


![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/129671265/1a91b443-725c-42a6-bb95-bdc0813121c4)

![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/129671265/d13ebc82-07a9-4792-9a89-472f1e328e87)
